joomla/joomla-cms
# Correct Cache setWorkarounds to take account of multiple script or st…

…yle declarations being used that mean that we have an array of scripts or stylesheets to be cached and not just a string

Cache setWorkarounds returns an array not a string

Pull Request for Issue # .

### Summary of Changes
Correct Joomla\CMS\Cache Cache setWorkarounds to take account of multiple script or style declarations being used that mean that we have an array of scripts or stylesheets to be cached and not just a string
Joomla\CMS\Cache Cache setWorkarounds returns an array not a string

### Testing Instructions

1. On a Joomla 4-dev site install this module https://www.yoursites.net/testextensions/mod_cachetest.zip
2. Create **2** instances of this module to display on the same page in different module positions - leave caching setting as 'global' in module settings
3. Enable Joomla conservative caching  in Joomla config
4. Enable maximum error reporting in Joomla config#
5. Make sure the Joomla cache is cleared
6. Load the frontend page where modules are displayed
7. Reload the frontend page

### Actual result BEFORE applying this Pull Request

At Step 6 you get a php warning on arguments to mb_strlen and 4 javascript alert messages

See https://www.yoursites.net/testextensions/Screenshot%202021-11-17%20at%2010-05-05%20Home.png

At step 7 you get a different PHP warning from libraries/src/Document/HtmlDocument.php because it expects an array but has been given a script by Cache.php

### Expected result AFTER applying this Pull Request

Page is displayed with no warning messages and module content is delivered from the cache correctly with 4 alerts

### Documentation Changes Required
